src/cache/src/index.ts

```diff
@@ -54,28 +56,25 @@ export type BooleanOrVoid = boolean | void
 const hash = (s: string) =>
   createHash('sha512').update(s).digest('hex')
 
+const success = <T>(p: Promise<T>): Promise<T | null> =>
+  p.then(
+    result => result,
+    () => null,
```

Comment:
Kind of a style suggestion, but it might be good to have something like:

```ts
const FAILURE = null
const success = <T>(p: Promise<T>): Promise<T | typeof FAILURE> =>
  p.then(
    result => result,
   () => FAILURE,
  )
```

Then, later on, you could do `if (FAILURE === success(...))` and it's slightly more obvious why `null` is being used there.